rpc/src/services/base_services.rs

```diff
@@ -39,6 +39,14 @@ pub enum RpcServiceError {
     StorageError { error: StorageError },
     #[fail(display = "No data found error, reason: {:?}", reason)]
     NoDataFoundError { reason: String },
+    #[fail(display = "Invalid or out-of-range datetime")]
+    TimestampOutOfRangeError,
+}
+
+impl From<TimestampOutOfRangeError> for RpcServiceError {
+    fn from(_: TimestampOutOfRangeError) -> Self {
```

Comment:
Can you also add the error itself to the resulting one? That is cleaner even if the underlying error is a unit structure, like this one.

Reply:
Ok can do.